src/Paramore.Brighter.MessagingGateway.AzureServiceBus/AzureServiceBusMessageProducer.cs

```diff
@@ -248,21 +248,40 @@ private async Task<IServiceBusSenderWrapper> GetSenderAsync(string topic)
         }
     }
 
-    private static ServiceBusMessage ConvertToServiceBusMessage(Message message)
+    private static ServiceBusMessage ConvertToServiceBusMessage(Message message, AzureServiceBusPublication publication)
     {
-        var cloudEvent = new CloudEvent(message.Header.Source.ToString(),
-            message.Header.Type,
-            new BinaryData(message.Body.Value))
+        BinaryData payload;
+        if (publication.UseCloudEvents)
         {
-            Id = message.Header.MessageId,
-            DataSchema = message.Header.DataSchema?.ToString(),
-            Subject = message.Header.Subject,
-            Time = message.Header.TimeStamp,
-            DataContentType = message.Header.ContentType
-        };
-        
-        
-        var azureServiceBusMessage = new ServiceBusMessage(new BinaryData(cloudEvent));
+            var dataFormat = CloudEventDataFormat.Binary;
+            if (!string.IsNullOrEmpty(message.Header.ContentType) &&
```

Comment:
The content type "application/cloudevents+json" should only apply if we use structured and NOT Binary. This is because in structure, we need to know that the body will be in envelope/message format and inline the headers followed by the content.

If we are binary (and this should be our default), then we only need to set the relevant headers and our content type is untouched.